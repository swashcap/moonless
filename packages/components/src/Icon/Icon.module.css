.Icon {
  line-height: 1;
  height: 1.5rem;
  width: 1.5rem;
}

.fillCurrentColor {
  fill: currentColor;
}

.fillNone {
  fill: none;
}

.strokeCurrentColor {
  stroke: currentColor;
}

.strokeNone {
  stroke: none;
}
