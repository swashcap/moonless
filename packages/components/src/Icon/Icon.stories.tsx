import {
  ArrowDownIcon,
  ArrowLeftIcon,
  ArrowRightIcon,
  ArrowUpIcon,
  CaretDownIcon,
  CaretLeftIcon,
  CaretRightIcon,
  CaretUpIcon,
  CheckIcon,
  ClearIcon,
  EditIcon,
  InfoIcon,
  MapPinIcon,
  MenuIcon,
  MinusIcon,
  PlusIcon,
  SuccessIcon,
  TagIcon,
  WarningIcon,
  XIcon,
} from './Icon';

export default {
  title: 'Component/Icon',
};

export const ArrowDown = () => <ArrowDownIcon />;
export const ArrowLeft = () => <ArrowLeftIcon />;
export const ArrowRight = () => <ArrowRightIcon />;
export const ArrowUp = () => <ArrowUpIcon />;
export const CaretDown = () => <CaretDownIcon />;
export const CaretLeft = () => <CaretLeftIcon />;
export const CaretRight = () => <CaretRightIcon />;
export const CaretUp = () => <CaretUpIcon />;
export const Check = () => <CheckIcon />;
export const Clear = () => <ClearIcon />;
export const Edit = () => <EditIcon />;
export const Info = () => <InfoIcon />;
export const MapPin = () => <MapPinIcon />;
export const Menu = () => <MenuIcon />;
export const Minus = () => <MinusIcon />;
export const Plus = () => <PlusIcon />;
export const Success = () => <SuccessIcon />;
export const Tag = () => <TagIcon />;
export const Warning = () => <WarningIcon />;
export const X = () => <XIcon />;
