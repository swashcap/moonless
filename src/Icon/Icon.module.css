.Icon {
  line-height: 1;
  height: 1.5rem;
  width: 1.5rem;
}
