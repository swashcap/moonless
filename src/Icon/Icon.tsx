import React from 'react';

import { clsx } from '../utils/clsx';
import styles from './Icon.module.css';

export type IconProps = React.SVGProps<SVGSVGElement>;

const Icon: React.FC<IconProps> = ({ className, ...rest }) => (
  <svg
    className={clsx(styles.Icon, className)}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 12 12"
    {...rest}
  />
);

export const CheckIcon: React.FC<IconProps> = (props) => (
  <Icon {...props}>
    <path
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      d="M2.5 6.5l2.5 3 4.5-6"
    />
  </Icon>
);

export const MinusIcon: React.FC<IconProps> = (props) => (
  <Icon {...props}>
    <path stroke="currentColor" stroke-linecap="round" d="M2 6h8" />
  </Icon>
);

export const XIcon: React.FC<IconProps> = (props) => (
  <Icon {...props}>
    <path stroke="currentColor" strokeLinecap="round" d="M3 3l6 6M3 9l6-6" />
  </Icon>
);
