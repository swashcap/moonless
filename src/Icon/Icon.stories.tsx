import React from 'react';

import { XIcon } from './Icon';

export default {
  title: 'Icon',
};

export const X = () => <XIcon />;
