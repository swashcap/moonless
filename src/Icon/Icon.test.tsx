import React from 'react';
import { render } from '@testing-library/react';

import { CheckIcon, XIcon } from './Icon';

describe('Icon', () => {
  describe('CheckIcon', () => {
    test('It should render.', () => {
      const { container } = render(<CheckIcon />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('XIcon', () => {
    test('It should render.', () => {
      const { container } = render(<XIcon />);

      expect(container).toMatchSnapshot();
    });
  });
});
