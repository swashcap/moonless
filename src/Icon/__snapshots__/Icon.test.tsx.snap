// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon CheckIcon It should render. 1`] = `
<div>
  <svg
    class="Icon"
    viewBox="0 0 12 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.5 6.5l2.5 3 4.5-6"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`Icon XIcon It should render. 1`] = `
<div>
  <svg
    class="Icon"
    viewBox="0 0 12 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 3l6 6M3 9l6-6"
      stroke="currentColor"
      stroke-linecap="round"
    />
  </svg>
</div>
`;
